src/StoreApi/Utilities/OrderController.php

```diff
@@ -431,15 +431,54 @@ protected function validate_coupon_usage_limit( \WC_Coupon $coupon, \WC_Order $o
 		$coupon_usage_limit = $coupon->get_usage_limit_per_user();
 
 		if ( $coupon_usage_limit > 0 ) {
-			$data_store  = $coupon->get_data_store();
-			$usage_count = $order->get_customer_id() ? $data_store->get_usage_by_user_id( $coupon, $order->get_customer_id() ) : $data_store->get_usage_by_email( $coupon, $order->get_billing_email() );
+			$data_store = $coupon->get_data_store();
+			// First, we check a logged in customer usage count, which happens against their user id, billing email, and account email.
+			if ( get_current_user_id() ) {
```

Comment:
Why this vs original $order->get_customer_id()?

Reply:
No strong reason, I think `get_customer_id` is better actually as it works if you're placing an order on someones name. Will fix this.

Reply:
[014bce6](https://github.com/woocommerce/woocommerce-blocks/pull/11860/commits/014bce602f61349e9edcc66865b2607d058b948f)